### DISCUSSION TITLE
[Chapter 06] View

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#49] [Chapter 06. Building a Web Application Server] [answered]
1. MVC에서 View의 정의
`이때 데이터를 표현하는 기능을 뷰(view)라고 하 며, 데이터를 처리하는 기능은 o.s.web.servlet.ViewResolver 인터페이스 가 담당한다. ViewResolver 구현체에 따라 데이터를 HTML, JSON, XML 등 다양한 형태로 표현할 수 있다.`
다들 아시는 것처럼 JSON은 데이터 포맷입니다. 그런데 이 문구를 보면 JSON도 그럼 Spring 안에서만큼은 View라고 봐도 된다는 것으로 이해하면 될까요?
그런데 밑에 나오는 문구에는 조금 관련이 없다고 생각하실 수도 있겠지만
`o.s.web.method.support.HandlerMethodReturnValueHandler 인터페이스는 컨트롤러 클래스에 정의된 핸들러 메서드가 리턴하는 객체를 변환하는 역할을 한다. 보통 View 기술을 사용하는 웹 애플리케이션에서 사용한다. REST-API 애플리케이션에서는 사용하는 일이 드물다. REST-API 애플리케이션에서는 JSON 타입의 메시지를 응답하고, 이 과정은 HttpMessageConverter로 변환되기 때문이다.`
라고 나옵니다. View와 Rest-api를 분리한 느낌인데 역시 View라고 보면 안되는게 맞을지 확실히 각인시키고 싶습니다.

2. @ModelAttribute
이건 질문글은 아니지만 저번에 스터디에 ModelAttribute에 관한 이슈가 제 나름대로 있었어서
찾아보니 
Request Body:
  - type: application/x-www-form-urlencoded
  - data: {

    }

form data형식으로 REST-API에서도 충분히 가능한걸로 확인했습니다!

**Comment by corock** [comment, marked as answer]
1. 책에서도 언급되듯 데이터를 처리하는 기능과 데이터를 표현하는 기능이 분리되어 있기 때문에
JSON 은 `ViewResolver` 를 통해 처리될 형태를 말하는 것이 아닐까요?

2. 집요하게 확인하는 모습에서 개발자의 혼을 느낄 수 있었습니다 😄
